webui-src/app/login.js

```diff
@@ -39,30 +45,39 @@ const loginComponent = function() {
             src: '../data/retroshare.svg',
             alt: 'retroshare_icon'
           }),
-          m('input[autofocus]', {
+          m('input' + (uname == '' ? '[autofocus]':''), {
             type: 'text',
+            value: uname,
             placeholder: 'Username',
             onchange: e => (uname = e.target.value)
           }),
-          m('input', {
+          m('input'+ (uname != '' ? '[autofocus]':''), {
             type: 'password',
             placeholder: 'Password',
-            onchange: e => (passwd = e.target.value)
+            onchange: e => (passwd = e.target.value),
+            onkeyup: e => {if (e.code=='Enter') loginBtn.click();}
           }),
           m('.extra', [
             'Port:',
             m('input', {
               type: 'number',
               value: port,
-              oninput: e => {
-                port = e.target.value;
-              }
+              oninput: e => (port = e.target.value)
+            })
+          ]),
+          m('.extra',[
+            'Server:',
+            m('input',{
+              type: 'text',
+              value: server,
+              oninput: e => (server = e.target.value)
             })
           ]),
           m(
             'button.submit-btn',
             {
-              onclick: () => verifyLogin(uname, passwd, port)
+              id: 'loginBtn',
+              onclick: () => verifyLogin(uname, passwd, port, 'http://' + server+':')
```

Comment:
It might be better to do the `http://` and `:` part inside `verifyLogin` itself, that way we can change `API_URL` also.

And I think we also need to check if user has already entered `http://` in the input.

Reply:
done